lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -50,8 +50,13 @@ def build(namespace:, context:, definition:, logger:, statsd_tags:, crd: nil)
         if crd
           CustomResource.new(crd: crd, **opts)
         else
-          inst = new(**opts)
-          inst.type = definition["kind"]
+          type = definition["kind"]
+          inst = if globals.include?(type.downcase)
+            KubernetesDeploy::GlobalKubernetesResource.new(**opts)
```

Comment:
What if instead of having a new class, we make `global` settable, and handle it like we do `type` below? I think might be a bit simpler overall.

Reply:
I went back and forth on this couldn't decide which was better. This felt more future proof, though I can't think of any special behavior we'd want from a GlobalResource., but the approach of not using a new class was definitely simpler. I'll try the other approach and see  how it goes.

Reply:
🤔 If we considered server resource lists part of the task's global operating context and put them into our new task config object, then we'd have a third option that might read even more straightforwardly: inside the normal superclass, we could do essentially `def global?; @task_config.global_resource_kinds.include?(kind); end`.

Reply:
I think that makes sense to do in https://github.com/Shopify/kubernetes-deploy/issues/572.